clients/cypress-e2e/README.md

```diff
@@ -0,0 +1,25 @@
+# Cypress E2E Tests
+
+This folder is meant to contain true end to end tests for Fides applications. Unlike Cypress tests nested within `admin-ui` or `privacy-center`, these tests do not stub any of their endpoints, and instead they require running against the Fides test environment set up.
+
+## Running
+
+First, start up the test environment. This will spin up all relevant servers and frontend services. From the root directory of this repo:
+
+```
+nox -s test_env
+```
+
+Admin UI will be found at `localhost:3000` and Privacy Center at `localhost:3001`.
```

Comment:
This is maybe more of a limitation of `test_env`, but testing against `localhost:3000` is less than ideal:
 - Pages will have inconsistent (slow) load time because it runs the dev server, so the dev server could introduce failures (false negative).
 - API requests will be happening through Next's redirects, so the dev server could hide something that's really a failure (false positive).
 - Feature flags will come from `NEXT_PUBLIC_APP_ENV="development"`.

If we're requiring a build before running the tests, then the static app served on `localhost:8080` should 🤞 be the Production™️ version we really want to test against.

Reply:
okay it took me a bit to wrap my head around how `test_env` works, but it looks like we can make this happen with [two small but potentially consequential changes](https://github.com/ethyca/fides/compare/aking/1293/cypress-test-env...aking/1293/cypress-prod-test-env).

1. Change the `test_env` image to build the prod image instead of the dev image. This will build the admin UI such that it can be served on `localhost:8080`
2. Add an exception to our volume mount so that it does not overwrite the built frontend assets

However, I imagine there was a reason to use the `dev` image for `test_env`. I think using the prod image instead of the dev image makes it so the package won't actually reflect dev changes (maybe because it doesn't get the `-e` editable flag?), so making this change would probably mess with other developers' flows.

I _think_ 2 is safe to do regardless, and might even prevent some common mistakes, but I haven't thought through all the use cases yet.

One thing we could do is add a totally new command, something like `nox -s prod_test_env` which uses the prod image instead of the dev image, and then we use that one for cypress instead of the current `test_env`. Might also address this old issue https://github.com/ethyca/fides/issues/928

This might be more of a question of what `test_env` _should_ be. Should it mimic a prod env as much as possible, or should it be a developer test environment? @NevilleS , any thoughts on how to move forward here?

Reply:
I'd prefer to have `test_env` be a production environment myself. You can get some additional feedback from the eng team to see how they are using it, but the more realistic that environment is the better it'll work for us.

Note that it'll also be faster to test with and should be less of a resource hog if you tighten up that test environment to use fewer containers and production images, which is a bonus.

Reply:
One note about this commit though @allisonking: https://github.com/ethyca/fides/compare/aking/1293/cypress-test-env...aking/1293/cypress-prod-test-env

That `build(test)` session is good but note that it doesn't force a rebuild of the `fides-privacy-center` image, which you'll also want for this.

Reply:
See https://github.com/ethyca/fides/blob/67a50d16d97030d9bd7afb18b417db68209dc8be/noxfiles/docker_nox.py#L102

That could just be changed to also include the `test` target, and probably already should have!

Reply:
thanks! made a separate ticket for changing all that up so we can do it separate from this PR: https://github.com/ethyca/fides/issues/2385

Reply:
ah, it loads so much faster now! hopefully will be more consistent, too. ez [change](https://github.com/ethyca/fides/pull/2241/commits/b765637d1746bddccbea56812e007547ec04f56a) once https://github.com/ethyca/fides/pull/2399 went in